snapshots/overlay/overlay.go

```diff
@@ -77,6 +78,15 @@ func WithMountOptions(options []string) Opt {
 	}
 }
 
+// WithMetaStore allows the MetaStore to be created outside the snapshotter
+// and passed in.
+func WithMetaStore(ms *storage.MetaStore) Opt {
```

Comment:
Directory structure managed by overlayfs snapshotter (e.g. `upperPath()`) should also be observed by remote snapshotter. Otherwize remote snapshotter needs to be implemented with assuming the directory structure and possibly can't notice the changes on the structure when bumping up the imports.

Reply:
Are you suggesting we make `upperPath` and `workPath` public, which means we’ll also have to make the `snapshotter` struct public?